app/main.py

```diff
@@ -1,3 +1,66 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: int) -> Distance :
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        if isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        return NotImplemented
+
+    def __iadd__(self, other: int) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (int, float)):
+            self.km += other
+        else:
+            return NotImplemented
+        return self
+
+    def __mul__(self, other: int) -> Distance:
+        if isinstance(other, (int, float)):
+            return Distance(self.km * other)
+        return NotImplemented
+
+    def __truediv__(self, other: int) -> Distance:
```

Comment:
The `__truediv__` method's type annotation for `other` should be `Union[int, float]` to reflect that it can accept numeric types. Additionally, consider raising a `ZeroDivisionError` if `other` is zero to handle division by zero explicitly.